### ISSUE TITLE
Plugin Configuration: JSON editor breaks named capturing groups in regular expressions

### ISSUE THREAD

**Issue by jensschuppe** [Project60/org.project60.banking#307]
Regexp patterns like
```json
"pattern": "\/Multi-Nr (?P<mfk>[0-9]{10})\/",
```
in a plugin are being destroyed when the configuration is being displayed with the JSON editor, which is stripping the named capturing group's name:

```json
"pattern": "\/Multi-Nr (?P[0-9]{10})\/",
```

If you save the configuration via the UI without restoring the capturing groups' names, the plugin won't work anymore, obviously.

This may be an upstream issue, but the [editor](http://jsoneditoronline.org/) itself does not show this behavior.

**Comment by bjendres** [comment]
Cannot reproduce. Could somebody give detailed instructions?

**Comment by jensschuppe** [comment]
Reproducible with

* CiviCRM 5.29.1
* CiviBanking 0.7-alpha13

It's not the JSON editor, the value in the hidden form field already contains the faulty code.

We should try the latest CiviBanking version and maybe try to restrict the behavior to specific CiviCRM versions.

**Comment by jensschuppe** [comment]
Some other things are being reformatted, such as `>` and `<`, which are being converted to HTML entities `&gt;` and `&lt;` - those might break `calculate` expressions that will be run through PHP's `eval()'  internally.

**Comment by bjendres** [comment]
released with ``0.8.0``
